src/profile-logic/marker-schema.js

```diff
@@ -432,6 +445,14 @@ export function formatFromMarkerSchema(
     case 'string':
       // Make sure a non-empty string is returned here.
       return String(value) || '(empty)';
+    case 'unique-string':
+      if (stringTable !== null && stringTable !== undefined) {
+        return stringTable.getString(value);
+      }
+      console.warn(
+        `Could not find marker table for index "${value}" in markertype "${markerType}"`
+      );
+      return 'Null or undefined string table.';
```

Comment:
If we go with removing the optional-ness of this string table, we can remove this string table check and warnings etc as well.

I think it's better to pass this string table all the time even though we are sure that we are not going to need it. It's also good for type checking. For example, when I made it non-optional, I found out a place that we want to update as well:
https://github.com/firefox-devtools/profiler/blob/e38bc75cd1f07c1f3dff2e4e3a839d289030ede6/src/profile-logic/marker-schema.js#L415-L418
This is a recursive call inside the `formatFromMarkerSchema` itself and we need to pass the string table to here as well. It's either `format` or 'string' and `format` can be a `unique-string` as well for this case.

Reply:
Agreed. I think the optional-ness was from an earlier design, which should really have been made mandatory while I was refactoring.